### PR TITLE
feat: handling same as Notion when user fills in the mixing of number and text

### PR DESCRIPTION
**Pull request by 0xkelvin**
@appflowy could you please help me review this change
i tried to modify the code to achieve the same behaviours like Notion app for Number type.

the only problem is when i fill in the text under Number type, the backend in Rust processes as an empty string which is correct, But the frontend still display the string on the cell instead of empty the cell.

### CODE REVIEW

File: frontend/rust-lib/flowy-grid/src/services/grid_editor.rs [hunk 1]
```diff
@@ -510,6 +510,7 @@ impl GridRevisionEditor {
                 // Update the changeset.data property with the return value.
                 let type_cell_data =
                     apply_cell_data_changeset(cell_changeset, cell_rev, field_rev, Some(self.cell_data_cache.clone()))?;
+                tracing::trace!("xxxxxxxxxxxx:: {}", type_cell_data);
```
Comment (appflowy) [inline frontend/rust-lib/flowy-grid/src/services/grid_editor.rs:513]: Please remove this.

File: frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs [hunk 2]
```diff
@@ -97,10 +98,14 @@ impl NumberTypeOptionPB {
 
     pub(crate) fn format_cell_data(&self, s: &str) -> FlowyResult<NumberCellData> {
         match self.format {
-            NumberFormat::Num => match Decimal::from_str(s) {
-                Ok(value, ..) => Ok(NumberCellData::from_decimal(value)),
-                Err(_) => Ok(NumberCellData::new()),
-            },
+            NumberFormat::Num => {
+                let re = Regex::new(r"[^\d\.]").unwrap();
```
Comment (appflowy) [inline frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs:102]: You can try lazy_static. For example:
```
lazy_static! {
    static ref URL_REGEX: Regex = Regex::new(
        "[(http(s)?):\\/\\/(www\\.)?a-zA-Z0-9@:%._\\+~#=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)"
    )
    .unwrap();
}
```

File: frontend/rust-lib/flowy-grid/src/services/grid_editor.rs (hunk 1 above)
Comment (0xkelvin) [reply frontend/rust-lib/flowy-grid/src/services/grid_editor.rs]: removed

File: frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs (hunk 2 above)
Comment (0xkelvin) [reply frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs:102]: ah, thanks for suggestion, this is the good point, i was consider lazy_static, but then i think the Regex is not to expensive to compute, so we don't really need initialized lazily. i think so.

Comment (appflowy) [reply frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs:102]: That depends. 
```
#[cfg(test)]
mod tests {
    use fancy_regex::Regex;
    use std::time::Instant;

    #[test]
    fn time_cost_test() {
        let start = Instant::now();
        let re = Regex::new(r"[^\d\.]").unwrap();
        let duration = start.elapsed();
        println!("Time elapsed in expensive_function() is: {:?}", duration);
    }
}

```
The result is `3ms` test on my computer.
image


So if we want to get 1000 cells' content using the `format_cell_data`, that will be expensive.

Comment (0xkelvin) [reply frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs:102]: nice, i see, let me switch to use lazy_static!

Comment (0xkelvin) [reply frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs]: done, @appflowy please help me review again

**Comment by 0xkelvin** [comment]
fixed : #1603

**Pull request merged**